### PR TITLE
HPCC-24121 NFS Helm chart example

### PR DESCRIPTION
**Pull request by jakesmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - please review

File: dockerfiles/examples/nfs/values-storage-nfs.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Overrides for azure storage in hpcc, using custom StorageClass'
+# NB: The "retained-storage" helm charts should be installed 1st.
```
Comment (richardkchapman) [inline dockerfiles/examples/nfs/values-storage-nfs.yaml:2]: Should "retained-storage" say "nfs-server here? Should the existingClaim names reflect the storage type in use?

**Comment by jakesmith** [comment]
@richardkchapman @ghalliday - plrease review

**Comment by richardkchapman** [comment]
Typo in git comment asking us to review...

File: dockerfiles/examples/nfs/hpcc-nfs/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+apiVersion: v2
+name: hpcc-nfs
+description: A Helm chart to provision HPCC PVC's using an NFS deployment as shared storage
+
+# A chart can be either an 'application' or a 'library' chart.
```
Comment (richardkchapman) [inline dockerfiles/examples/nfs/hpcc-nfs/Chart.yaml:5]: Not sure these comments are useful

Comment (jakesmith) [reply dockerfiles/examples/nfs/hpcc-nfs/Chart.yaml:5]: Agree.
These are a result of helm create auto-generation (as is .helmignore), I trimmed some of the excess, but not sure where to stop.
e.g. should all the unused definitions in _helpers.tpl be removed too?

**Comment by jakesmith** [comment]
@ghalliday - I've adjusted a couple of comments in the README etc. , following changes on https://github.com/hpcc-systems/HPCC-Platform/pull/13800 

Please check.

File: dockerfiles/examples/nfs/hpcc-nfs/Chart.yaml (hunk 2 above)
Comment (richardkchapman) [reply dockerfiles/examples/nfs/hpcc-nfs/Chart.yaml:5]: These comments are notes to the developer writing a new chart, and serve no purpose once you have done so.

Yes, you probably should trim unused definitions in helpers too

File: dockerfiles/examples/nfs/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# HPCC NFS storage
+
+## Directory contents
+
+### hpcc-nfs/
+
+The hpcc-nfs helm chart will provision a NFS Server and optionally a default shared PVC using the default Storage Class.
```
Comment (richardkchapman) [inline dockerfiles/examples/nfs/README.md:7]: Perhaps explain why we might want to provision an NFS server in this way?

File: dockerfiles/examples/nfs/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+Example use:
+
+*helm install nfsstorage hpcc-nfs/*
+*helm install myhpcc hpcc/ --set global.image.version=XXX -f examples/local/values-nfs.yaml*
```
Comment (richardkchapman) [inline dockerfiles/examples/nfs/README.md:12]: Perhaps latest rather than XXX

Comment (richardkchapman) [reply dockerfiles/examples/nfs/README.md:12]: I'm guessing the * at start/end of these lines is to make the line appear bold when viewed in a md viewer, but it makes it very confusing if viewed as plain text. I suspect would be better without them (I notice they are not present in the localhost example)

**Comment by richardkchapman** [comment]
@jakesmith Some minor comments

**Comment by jakesmith** [comment]
@richardkchapman - addressed your comments, updated.

**Comment by richardkchapman** [comment]
@jakesmith Looks good, please squash

**Comment by jakesmith** [comment]
@richardkchapman - squashed

**Pull request merged**